experimental/traffic-portal/src/app/api/user.service.ts

```diff
@@ -267,6 +268,37 @@ export class UserService extends APIService {
 		return this.get<Array<ResponseRole>>(path).toPromise();
 	}
 
+	/**
+	 * Creates a new Role.
+	 *
+	 * @param role The role to create.
+	 * @returns The created role.
+	 */
+	public async createRole(role: RequestRole): Promise<ResponseRole> {
+		return this.post<ResponseRole>("roles", role).toPromise();
+	}
+
+	/**
+	 * Updates an existing Role.
+	 *
+	 * @param role The role to update.
+	 * @returns The updated role.
+	 */
+	public async updateRole(role: ResponseRole): Promise<ResponseRole> {
+		return this.put<ResponseRole>(`roles?name=${role.name}`, role).toPromise();
+	}
+
+	/**
+	 * Deletes an existing role.
+	 *
+	 * @param role The role to be deleted.
+	 * @returns The deleted role.
+	 */
+	public async deleteRole(role: string | ResponseRole): Promise<void> {
+		const roleName = typeof(role) === "string" ? role : role.name;
+		return this.delete(`roles?name=${roleName}`).toPromise();
+	}
```

Comment:
These functions are not covered by tests, which brings the test coverage for the user service down to 94.02% of statements from 100% of statements.

Reply:
I am curious, where are you seeing this % number?

Reply:
You can see a report of the coverage using the `--code-coverage` flag for `ng test`. So, to put that all together: `ng test --code-coverage`.